src/ElectronicRegulator.cpp

```diff
@@ -671,78 +116,59 @@ void setup() {
+    double LPpsi = utility::voltageToPressure(analogRead(LP_PT));
+    double InjectorPT = utility::voltageToPressure(analogRead(INJECTOR_PT));
+    double lastPrint = 0;
+    String inString = "";
 
-    angle = encoder.read();
-    motorAngle = encoderToAngle(angle);
-    potAngle = readPot();
-    HPpsi = voltageToHighPressure(analogRead(HP_PT));
-    LPpsi = voltageToPressure(analogRead(LP_PT));
-    InjectorPT = voltageToPressure(analogRead(INJECTOR_PT));
-    
     // LPpsi = analogRead(POTPIN)/1024.0*360;
 
-    dt=micros()-t2;
-    t2+=dt;
-    isAngleUpdate=(angle!=oldPosition);
-    e=angle-angle_setpoint;
+    // isAngleUpdate=(angle!=oldPosition);
+    // e=angle-angle_setpoint;
 
     //Compute Inner PID Servo loop
-    float rawSpd=-(kp*e+kd*(e-oldError)/float(dt));
-    if(rawSpd<MAX_SPD && rawSpd>MIN_SPD){ //anti-windup
-        angle_errorInt+=e*dt;
-        rawSpd-=ki*angle_errorInt;
-    }
-    else{angle_errorInt=0;}
-    
-    if (isAngleUpdate) {
-        oldPosition = angle;
-    }
-    oldError=e;
-
-    //Compute Outer Pressure Control Loop
-    pressure_e = LPpsi - pressure_setpoint;
-    p_buff->insert(t2/1.0e6, LPpsi);
-    double rawAngle = -( kp_outer*pressure_e + kd_outer*(p_buff->get_slope()) );
-    if(rawAngle<MAX_ANGLE && (rawAngle>MIN_ANGLE || pressure_errorInt<0)){
-        pressure_errorInt += pressure_e * dt;
-        rawAngle -= ki_outer * pressure_errorInt;
-    }
-    pressure_e_old = pressure_e;
+    long angle_setpoint = 0;
+    PID inner = PID (11.5, 1.5e-6, 0.1665e6, angle_setpoint, false);
+    speed = inner.update(angle);
 
-    // Constrain angles and speeds
-    angle_setpoint = min(MAX_ANGLE, max(MIN_ANGLE, rawAngle));
-    speed = min(max(MIN_SPD,rawSpd),MAX_SPD);
+    //Compute Outer Pressure Control Loop and constrain angles and speeds
+    double pressure_setpoint = 130;
+    PID outer = PID (30, 30.0e-6, 2.5, pressure_setpoint, true);
+    angle = outer.update(LPpsi);
 
-    runMotor();
+    utility::runMotor(speed);
+    double t2, error = outer.transmit();
 
     if (t2 - lastPrint > 1000) {
         #ifndef USE_DASHBOARD
         Serial.println( String(t2) + "\t"+ String(angle_setpoint) + "\t"+ String(pressure_setpoint) +"\t" + String(speed) + "\t" + String(motorAngle) + "\t" + String(HPpsi) + "\t" + String(LPpsi) + "\t" + String(InjectorPT) + "\t" + String(p_buff->get_slope()) + "\t" + String(pressure_errorInt) );     
         #else
         Comms::Packet packet = {.id = 1};
         // Comms::packetAddFloat(&packet, sin(t2/1e6));
-        Comms::packetAddFloat(&packet, float(angle_setpoint));
-        Comms::packetAddFloat(&packet, pressure_setpoint);
-        Comms::packetAddFloat(&packet, float(speed));
+        Comms::packetAddFloat(&packet, angle_setpoint);
+        Comms::packetAddFloat(&packet, pressure_setpoint); //not sure how to send, possibly store these variables in class when calling updatePressure method
```

Comment:
one way to do this is to have get_setpoint function for the pid class, and then call that